### PR TITLE
chore: bump deps and go version

### PR DESCRIPTION
**Pull request by JorTurFer**

### CODE REVIEW

**Comment by JorTurFer** [comment]
I think that the image still uses go 1.19

**Comment by dgrisonnet** [comment]
Seems like the test failure is do to golangci-lint running out of memory. We will need to look into disabling the plugin that causes this issue.

**Comment by JorTurFer** [comment]
> Seems like the test failure is do to golangci-lint running out of memory. We will need to look into disabling the plugin that causes this issue.

I forgot to update it and 1.50.1 doesn't support go 1.20, I have updated golangci and it works, but some issues have been detected, and I'm fixing them right now

File: pkg/apiserver/installer/conversion.go [hunk 1]
```diff
@@ -63,15 +63,7 @@ func ConvertURLValuesToV1beta2MetricListOptions(in *url.Values, out *cmv1beta2.M
...
 // RegisterConversions adds conversion functions to the given scheme.
 func RegisterConversions(s *runtime.Scheme) error {
-	if err := s.AddConversionFunc((*url.Values)(nil), (*cmv1beta1.MetricListOptions)(nil), func(a, b interface{}, scope conversion.Scope) error {
+	return s.AddConversionFunc((*url.Values)(nil), (*cmv1beta1.MetricListOptions)(nil), func(a, b interface{}, scope conversion.Scope) error {
 		return ConvertURLValuesToV1beta1MetricListOptions(a.(*url.Values), b.(*cmv1beta1.MetricListOptions), scope)
-	}); err != nil {
-		return err
-	}
-	if err := s.AddConversionFunc((*url.Values)(nil), (*cmv1beta2.MetricListOptions)(nil), func(a, b interface{}, scope conversion.Scope) error {
-		return ConvertURLValuesToV1beta2MetricListOptions(a.(*url.Values), b.(*cmv1beta2.MetricListOptions), scope)
```
Comment (dgrisonnet) [inline pkg/apiserver/installer/conversion.go:72]: I don't think we want to remove that, do we?

Comment (JorTurFer) [reply pkg/apiserver/installer/conversion.go:72]: you are right, we don't want, I thought that it was repited but they are different, nice catch!

**Review by dgrisonnet — APPROVED** [review]
Thanks @JorTurFer! :)

/lgtm
/approve

**Pull request merged**